packages/backend-common/src/reading/FetchUrlReader.ts

```diff
@@ -60,11 +69,29 @@ export class FetchUrlReader implements UrlReader {
               }
             : (_url: URL) => true;
           const host = allowConfig.getString('host');
-          if (host.startsWith('*.')) {
-            const suffix = host.slice(1);
-            return (url: URL) => url.host.endsWith(suffix) && checkPath(url);
+          const [hostname, port] = host.split(':');
+
+          let checkPort = (url: URL) => !url.port;
+          if (port) {
+            const isRange = port.includes('-');
+            const range = port.split('-').map(v => parseInt(v, 10)) as [
+              number,
+              number | undefined,
+            ];
+
+            // it allows port ranges ex: :1-200 or :30-
+            checkPort = isRange
+              ? (url: URL) => isInRange(parseInt(url.port, 10), range)
+              : (url: URL) => port === url.port;
```

Comment:
What do you think about supporting `http` -> port 80 and `https` -> port 443? So that if you specify `example.com:443` then `https://example.com` is allowed? Typically you probably won't use ranges that low, but figuring it would be a bit unexpected if that translation was not supported.